migration-engine/ARCHITECTURE.md

```diff
@@ -117,6 +118,41 @@ On a high level, migrate deploy _exclusively_:
   looking at the `_prisma_migrations` table in the target database.
 - Run the migrations that have not been applied yet, in chronological order.
 
+`prisma migrate deploy` is the command meant to be used for _unattended_
+migrations (as part of CI). As such, it should be as reliable, predictable and
+deterministic as possible.
+
+### Why does `prisma migrate deploy` not warn when a migration is in the migrations table but not in the migrations folder?
+
+We don't want to warn if an already applied migration is missing from the
+migration directory because it would prevent squashing migrations and
+deployment from not-properly-rebased branches.
+
+In general, by design, `deploy` errs on the side of not standing in the way of
+deploying your migrations.
+
+### Why does `prisma migrate deploy` not detect drift?
+
+We don't detect drift because we want to keep the deployment path light and
+simple, and because we would need a shadow database for that. Many people would
```

Comment:
There would also be other ways to detect a mismatch of production and the "before" state of the database the migration was created on if Migrate worked differently and had other information. This statement sounds a bit like a shadow database would be the only way, but comparing an introspected schema of the production database with a "before" schema (if that was available) would for example also be a way.

Maybe qualify this sentence better to show it means "the way Prisma Migrate currently works".

Reply:
>  comparing an introspected schema of the production database with a "before" schema (if that was available

The only way we have to obtain that "before" schema is exactly the shadow database. I don't see a reasonable alternative with the current architecture. I can qualify.

Reply:
Oh yeah, I know that how Migrate works this is not possible any other way. But that is just because how Prisma Migrate works, not universal truth 👍

Reply:
Totally!